### PR TITLE
MBS-10393 fix logger dateformatter concurrent problem

### PR DESCRIPTION
**Pull request by dsvoronin**
remove checker

### CODE REVIEW

File: subprojects/common/elastic/src/main/kotlin/com/avito/android/elastic/HttpElasticClient.kt [hunk 1]
```diff
@@ -96,7 +93,16 @@ internal class HttpElasticClient(
...
     }
 
-    private fun utcFormatter(pattern: String): DateFormat = SimpleDateFormat(pattern, Locale.getDefault()).apply {
-        timeZone = TimeZone.getTimeZone("UTC")
+    /**
+     * from javadoc:
+     *
+     * It is recommended to create separate format instances for each thread.
+     * If multiple threads access a format concurrently, it must be synchronized
+     * externally.
```
Comment (eugene-krivobokov) [inline subprojects/common/elastic/src/main/kotlin/com/avito/android/elastic/HttpElasticClient.kt:101]: Another good reason to use date-time API.
`java.time.format.TimeFormatter` is thread-safe.
We can use the original API in Gradle and desugar or backported version in Android code.

**Review by eugene-krivobokov — APPROVED** [review]

File: subprojects/common/elastic/src/main/kotlin/com/avito/android/elastic/HttpElasticClient.kt (hunk 1 above)
Comment (dsvoronin) [reply subprojects/common/elastic/src/main/kotlin/com/avito/android/elastic/HttpElasticClient.kt:101]: yep, but we have to evaluate desugaring impact on build times

File: subprojects/common/elastic/src/main/kotlin/com/avito/android/elastic/HttpElasticClient.kt [hunk 2]
```diff
@@ -96,7 +93,16 @@ internal class HttpElasticClient(
...
+     */
+    private fun utcFormatter(pattern: String) = object : ThreadLocal<DateFormat>() {
```
Comment (sergioboi) [inline subprojects/common/elastic/src/main/kotlin/com/avito/android/elastic/HttpElasticClient.kt:103]: ThreadLocal.withInitial { }

Comment (dsvoronin) [reply subprojects/common/elastic/src/main/kotlin/com/avito/android/elastic/HttpElasticClient.kt:103]: This code used in ui tests env on 22 API emulators:
`Call requires API level 26 (current min is 21): java.lang.ThreadLocal#withInitial`
and it is not supported for now even with desugaring: https://developer.android.com/studio/write/java8-support-table

**Pull request merged**